File: task2/main.swift
import Foundation

// Ввод количества команд
print("Введите количество команд: ", terminator: "")
guard let input = readLine(), var n = Int(input) else { exit(0) }

var matches = 0

// Рассчёт матчей по системе плей-офф
while n > 1 {
    if n % 2 == 0 {
        matches += n / 2  // Чётное количество команд
        n /= 2
    } else {
        matches += (n - 1) / 2  // Нечётное количество команд
        n = (n - 1) / 2 + 1
    }
}

print(matches)  // Вывод общего числа матчей

File: task2/main.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Введите количество команд: ");
        int n = int.Parse(Console.ReadLine());
        int matches = 0;

        while (n > 1)
        {
            if (n % 2 == 0)
            {
                matches += n / 2;
                n /= 2;
            }
            else
            {
                matches += (n - 1) / 2;
                n = (n - 1) / 2 + 1;
            }
        }

        Console.WriteLine(matches);
    }
}
